File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    
    String[] vowels = {"A", "E", "I", "O", "U"};
    int count = 0;
    boolean found = false;
    
    public int solution(String word) {        
        dfs("", word);
        
        return count;
    }
    
    void dfs(String current, String word) {
        
        if (found) return;
        if (current.length() > 5) return;
        
        if (current.equals(word)) {
            found = true;
            return;
        }
    
        for (String vowel : vowels) {
            dfs(current + vowel, word);
        }
        
        count++;
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
vowels = ['A', 'E', 'I', 'O', 'U'];
answer = 0
found = False

def solution(word):
    
    dfs("", word);
    
    return answer

def dfs(cur, word):
    global answer
    global found
    
    if (found):
        return;
    
    if (len(cur) > 5): 
        return;

    if (word == cur):
        found = True
        return
    
    for i in range(len(vowels)):
        dfs(cur + vowels[i], word)
        
    answer = answer + 1;